File: Lab-06-solution/src/main/java/com/petclinic/core/Visit.java
package com.petclinic.core;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;

import java.time.LocalDate;

@Entity
public class Visit {
    @Id @GeneratedValue
    private Long id;
    @Column(unique = true)
    private String referenceNumber;
    private LocalDate date;
    private String purpose;



    public Visit() {
    }

    public Visit(Long id, String referenceNumber, LocalDate date, String purpose) {
        this.id = id;
        this.referenceNumber = referenceNumber;
        this.date = date;
        this.purpose = purpose;
    }

    public Long getId() {
        return id;
    }

    public String getReferenceNumber() {
        return referenceNumber;
    }

    public LocalDate getDate() {
        return date;
    }

    public String getPurpose() {
        return purpose;
    }
}

File: spring-core-lab-solutions/Lab-kotlin-solution/src/main/kotlin/com/petclinic/core/Visit.kt
package com.petclinic.core

import jakarta.persistence.*
import jakarta.validation.constraints.NotBlank
import java.time.LocalDate

@Entity
data class Visit(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int?,
    val date: LocalDate,
    @NotBlank(message = "Pet name must not be blank")
    val purpose: String,
    @OneToOne(cascade = [CascadeType.PERSIST])
    @JoinColumn(name="owner_id")
    val owner: Owner,
    @OneToOne(cascade = [CascadeType.PERSIST])
    @JoinColumn(name="pet_id")
    val pet: Pet
)
